docs/development.md

```diff
@@ -0,0 +1,181 @@
+# Development
+
+## Structure
+
+This repository is a monorepo. Here's the list of all the apps and packages in it (each app has its own README):
+
+### Apps
+
+- [`apps/storefront`](../apps/storefront/README.md): Next.js e-commerce storefront starter pack
+- [`apps/saleor-app-checkout`](../apps/saleor-app-checkout/README.md): a Next.js Saleor app with a dashboard for managing settings and theme, backend for Checkout, and the Checkout Storefront – ready to be extended/modified
+
+### Packages
+
+- `packages/ui-kit`: UI kit for Storefront and Checkout
+- `packages/config-checkout`: `eslint` configuration
+- `packages/config-storefront`: `eslint` configuration
+- `packages/tsconfig`: `tsconfig.json`s used throughout the monorepo
+- `packages/checkout-storefront`: Checkout storefront component
+- `packages/checkout-common`: Common types and utils
+
+## Getting started
+
+### Prerequisites
+
+This monorepo uses `pnpm` as a package manager, so you will have to install it beforehand. To install it, run:
+
+```bash
+npm install -g pnpm
+```
+
+### Installing dependencies
+
+To install dependencies, run:
+
+```bash
+pnpm i
+```
+
+### Running
+
+To start all the dev servers, run:
+
+```bash
+pnpm dev
+```
+
+[The command will boot up 4 applications running on different ports.](#ports).
+
+You can also start a specific app by running:
+
+```bash
+pnpm dev --filter=saleor-app-checkout
+```
+
+#### [`saleor-app-checkout`](../apps/saleor-app-checkout/README.md)
+
+First, you need to change the default environment variables. Create `.env.local` file in `apps/saleor-app-checkout`, as described [here](../apps/saleor-app-checkout/README.md#env-variables).
+
+Then, to develop `saleor-app-checkout` app locally, you need to create a tunnel for it. The tunnel will enable you to display it within your Saleor Dashboard.
+
+If you want to read more about Saleor Apps architecture, see [the documentation](https://docs.saleor.io/docs/3.x/developer/extending/apps/key-concepts).
+
+To create a tunnel, run:
+
+```bash
+cd apps/saleor-app-checkout && npx saleor app tunnel 3001
+```
+
+> Note: The process needs to be running in the background. Please don't kill it during the development.
+
+<!-- is the bit below correct? you have to filter something to run it for all the apps? -->
+<!-- 
+To run the development server for all the apps, use the following command:
+
+```
+pnpm run dev --filter=saleor-app-checkout...
+```
```

Comment:
No this bit is not correct, you *can* use filter to run only `saleor-app-checkout` + it's dependencies but most of the time you want to run all dev servers. We can drop the `--filter` part.